src/python/pants/backend/jvm/targets/runtime_platform_mixin.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from abc import ABC
+
+from pants.backend.jvm.subsystems.jvm_platform import JvmPlatform
+from pants.base.payload_field import PrimitiveField
+
+
+class RuntimePlatformMixin(ABC):
+  """A mixin that identifies a target class as one that can have a jvm runtime_platform.
+
+  mixin in usage
+
+  change __init__:
+    - Add runtime_platform=None to the kwargs.
+    - Add the following to the doc string:
+    :param str runtime_platform: The name of the platform (defined under the jvm-platform subsystem)
+      to use for runtime (that is, a key into the --jvm-platform-platforms dictionary). If
+      unspecified, the platform will default to the first one of these that exist: (1) the
+      default_runtime_platform specified for jvm-platform, (2) the platform that would be used for
+      the platform kwarg.
+
+  :API: public
+  """
+
+  def init_runtime_platform(self, payload, runtime_platform):
```

Comment:
So, AFAIK, mixins are allowed to have constructors. So I think that rather than calling a method explicitly like this, you can get in the middle in the mixin's constructor and do the `payload = payload or Payload()` etc dance, and then pass the expanded payload along.

Having said that: you'll still need the docstring changes in all of the subclasses, because the docsite will only render the help for the the subclass.

Reply:
I'll do that. I think it's cleaner.

Reply:
The caveat with that is that the superclass order now matters. I added a note to the docstring about it.